dbhydra/dbhydra_core.py

```diff
@@ -789,6 +797,48 @@ def update(self, variable_assign, where=None):
         print(query)
         return self.db1.execute(query)
 
+    def update_from_df(
+            self, update_df: pd.DataFrame, where_column: Optional[str] = None, where_value: Any = None) -> None:
+        """Build UPDATE SQL query from a dataframe and execute it.
+
+        :param update_df: Dataframe with updated values - MUST only hold a single row
+        :type update_df: pd.DataFrame
+        :param where_column: Column name to use for WHERE clause.
+        :type where_column: str
+        :param where_value: Value to use for WHERE clause.
+        :type where_value: Any
+        """
+        if not len(update_df) == 1:
+            raise ValueError("There can only be one row in the UPDATE dataframe")
+
+        column_types_without_uid = self.types[1:]
```

Comment:
@DovaX 
I think this is principally wrong because we assume that the first column is always `id` (`int`), but all the classes have the optional `types` and `columns` parameters, which can be set arbitrarily

Reply:
True, I mistakenly assumed DBHydra always sets `id` column as a first one

Reply:
Introduced new commit, now `id` column is stripped based on `self.id_column_name`, or omitted if the following column does not exist.